src/main/scala/pirate.internal/ParseTraversal.scala

```diff
@@ -11,6 +11,10 @@ object ParseTraversal {
   case class ShortParsedName(c: Char) extends ParsedName
   case class ParsedWord(name: ParsedName, value: Option[String])
 
+  type P[A] = EitherT[PWR, ParseError, A]
```

Comment:
I think you will quickly find that you probably want to leave this as a data type wrapping the stack. Gives you somewhere to build out all the specific combinators you will want to make the stack usable in a neater way.